### PR TITLE
[Merged by Bors] - ATX: add API to differentiate publish and target epoch

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation

for code clarity, explicitly use PublishEpoch() vs TargetEpoch()

## Changes

- move TargetEpoch to challenge struct and add PublishEpoch API
- simplify StoreAtx API

### CODE REVIEW

File: activation/handler.go [hunk 1]
```diff
@@ -299,7 +297,7 @@ func (h *Handler) StoreAtx(ctx context.Context, ech types.EpochID, atx *types.Ve
 		delete(h.atxChannels, atx.ID())
 	}
 
-	h.log.WithContext(ctx).With().Info("finished storing atx in epoch", atx.ID(), ech)
+	h.log.WithContext(ctx).With().Info("finished storing atx in epoch", atx.ID(), atx.PublishEpoch())
```
Comment (fasmat) [inline activation/handler.go:300]: Should we use a formatted string here: "... atx %v in epoch %d" ?

File: activation/activation_test.go [hunk 2]
```diff
@@ -282,9 +282,7 @@ func assertLastAtx(r *require.Assertions, posAtx, prevAtx *types.VerifiedActivat
 }
 
 func storeAtx(r *require.Assertions, atxHdlr *Handler, atx *types.VerifiedActivationTx, lg log.Log) {
```
Comment (fasmat) [inline activation/activation_test.go]: should we inline this function? `storeAtx(r, atxHdlr, atx, nil)` is now just 

`r.NoError(atxHdlr.StoreAtx(context.TODO(), atx))`

**Review by fasmat — APPROVED** [review]

File: activation/handler.go (hunk 1 above)
Comment (countvonzero) [reply activation/handler.go:300]: that's inconsistent with the current practice.
these labeled field is searchable in loki. whereas formatted string can only be regex'ed when searching through log.
Dmitry and i were moving away from the formatted string in logging.

File: activation/activation_test.go (hunk 2 above)
Comment (countvonzero) [reply activation/activation_test.go]: done.

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**